### ISSUE TITLE
Support Arm64 in ResolveAssemblyReferences

### ISSUE THREAD

**Issue by rainersigwald** [microsoft/msbuild#3746]
If you attempt to use a .winmd with an implementation DLL in native ARM64, you get an error like

```
Unknown processor architecture. The implementation file "whatever.dll" for "whatever.winmd" had an ImageFileMachine value of "0xAA64". If you wish to use this implementation file make sure the "ResolveAssemblyWarnOrErrorOnTargetArchitectureMismatch" property in your project is set to "Warning" or "None".
```

We should add to the list of known architectures here:

https://github.com/Microsoft/msbuild/blob/3b3c3ea645cc529120c762cf2a8504ff960f7fb2/src/Tasks/AssemblyDependency/ReferenceTable.cs#L2773-L2794

Note: there's no corresponding entry in SystemProcessorArchitecture, so implementation may not be totally trivial.

**Comment by rainersigwald** [comment]
Note: we're also misclassifying the target arch of the output assembly:

https://github.com/Microsoft/msbuild/blob/3b3c3ea645cc529120c762cf2a8504ff960f7fb2/src/Tasks/AssemblyDependency/ResolveAssemblyReference.cs#L2907-L2935

It'll read MSIL when we ask for ARM64.

**Comment by rainersigwald** [comment]
Fixed by #4389
